graphql-dgs-codegen-shared-core/src/main/kotlin/com/netflix/graphql/dgs/client/codegen/GraphQLQueryRequest.kt

```diff
@@ -26,12 +26,13 @@ import graphql.schema.Coercing
 class GraphQLQueryRequest(
     val query: GraphQLQuery,
     val projection: BaseProjectionNode?,
-    scalars: Map<Class<*>, Coercing<*, *>>?
+    val inputValueSerializer: InputValueSerializer
 ) {
 
     constructor(query: GraphQLQuery) : this(query, null, null)
     constructor(query: GraphQLQuery, projection: BaseProjectionNode?) : this(query, projection, null)
-    val inputValueSerializer = InputValueSerializer(scalars ?: emptyMap())
+    constructor(query: GraphQLQuery, projection: BaseProjectionNode?, scalars: Map<Class<*>, Coercing<*, *>>?) : this(query, projection, InputValueSerializer(scalars ?: emptyMap()))
```

Comment:
I think you'll also need one constructor that had the original arguments, i.e. query, projection, scalars. Java does not have optional input like kotlin, so it will break those usages.

Reply:
On second thought, should we just take a boolean flag like `serializeAsNull` and instantiate the appropriate serializer accordingly? That way we don't expose the serializer interface to users, and we can instantiate the appropriate one?

Reply:
Line 34 should be the original constructor I am adding, is there a problem with how I am describing it? 

I don't mind either approach for how we enable this behavior, whether to open the interface or to provide a boolean flag. I would say it depends on how customizable we want this library to be, let me know if you have a preference.

Reply:
Ah true, I misread that. You do have it on line #34.

I was leaning towards not providing too much customizability, because opening up the interface introduces the need to make breaking changes if we change how things are serialized potentially in the future.

I don't have a strong opinion though, so we can leave it the way you have as well. Just some food for thought.

Reply:
Since we are both not strongly opinionated on the implementation detail, but prefer to reduce the amount of customization, I'll wrap the serializer injection in a constructor so that it doesn't expose itself.